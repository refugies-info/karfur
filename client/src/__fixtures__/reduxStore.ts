import { RootState } from "services/rootReducer";
import { SearchResultsState } from "services/SearchResults/searchResults.reducer";
import { UserFavoritesState } from "services/UserFavoritesInLocale/UserFavoritesInLocale.reducer";
import { activeThemesMock } from "./activeThemes";

const initialMockUserState = {
  user: null,
  admin: false,
  traducteur: false,
  expertTrad: false,
  contributeur: false,
  membreStruct: false,
  userId: "",
  userFetched: false,
  rolesInStructure: [],
};
const initialMockLangueState = {
  langues: [],
  languei18nCode: "fr",
  showLanguageModal: false,
  showLangModal: false,
};

const initialMockTranslationState = {
  translation: {
    initialText: {},
    translatedText: {},
  },
  translations: [],
};

const initialUserFavoritesState: UserFavoritesState = {
  favorites: [],
  showFavoriteModal: false
}

const initialMockSearchReults: SearchResultsState = {
  results: {
    dispositifs: [],
    demarches: [],
    dispositifsSecondaryTheme: []
  },
  query: {
    search: "",
    departments: [],
    themes: [],
    needs: [],
    age: [],
    frenchLevel: [],
    language: [],
    sort: "date",
    type: "all",
  }
}

export const initialMockStore = {
  user: initialMockUserState,
  langue: initialMockLangueState,
  activeDispositifs: [],
  tts: { ttsActive: false, showAudioSpinner: false },
  userStructure: null,
  selectedDispositif: null,
  loadingStatus: {},
  translation: initialMockTranslationState,
  activeStructures: [],
  selectedStructure: null,
  allDispositifs: [],
  allStructures: [],
  users: [],
  userFavorites: initialUserFavoritesState,
  userContributions: [],
  needs: [],
  themes: {
    activeThemes: activeThemesMock,
    inactiveThemes: []
  },
  searchResults: initialMockSearchReults
};
