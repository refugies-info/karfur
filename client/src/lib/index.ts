export const jsUcfirst = (string: string) =>
  string &&
  string.length > 1 &&
  string.charAt(0).toUpperCase() + string.slice(1, string.length);

export const jsLcfirst = (string: string) =>
  string &&
  string.length > 1 &&
  string.charAt(0).toLowerCase() + string.slice(1, string.length);

export const limitNbCaracters = (string: string, nbCaractersMax: number) =>
  string.substring(0, Math.min(string.length, nbCaractersMax)) +
  (string.length > nbCaractersMax ? "..." : "");

export const removeAccents = (str = "") => {
  var accents =
    "ÀÁÂÃÄÅàáâãäåÒÓÔÕÕÖØòóôõöøÈÉÊËèéêëðÇçÐÌÍÎÏìíîïÙÚÛÜùúûüÑñŠšŸÿýŽž";
  var accentsOut =
    "AAAAAAaaaaaaOOOOOOOooooooEEEEeeeeeCcDIIIIiiiiUUUUuuuuNnSsYyyZz";
  const chars = str.split("");
  let i, x;
  for (i = 0; i < chars.length; i++) {
    if ((x = accents.indexOf(chars[i])) !== -1) {
      chars[i] = accentsOut[x];
    }
  }
  return chars.join("");
};

// rules for password check :
// nbNumUpper = nbNumeric + nbUpper
// score 0 (cannot set password): length < 5
// score 0.5 : 7>length>=5 + nbNumUpper = 1 or 2 or0
// score 1.5 : 7>length >= 5 + nbNumUpper>=3 OR length>=7 + nbNumUpper<3
// score 2.5 : length >= 7 + nbNumUpper>=3
// score 3.5 : length >= 7 + nbNumUpper>=3 + at least 1 special caractere

// make sure to use same function in backend
export const computePasswordStrengthScore = (
  password: string
): { score: number } => {
  if (password.length < 5) {
    return { score: 0 };
  }

  const numerics = password.match(/\d+/g)?.join("");
  const nbNumeric = numerics ? numerics.length : 0;

  const upperCase = password.match(/[A-Z]/g)?.join("");
  const nbUpperCase = upperCase ? upperCase.length : 0;

  if (password.length < 7) {
    if (nbUpperCase + nbNumeric < 3) {
      return { score: 1 };
    }
    return { score: 2 };
  }

  if (nbNumeric + nbUpperCase < 3) {
    return { score: 2 };
  }
  const specialCar = password.match(/[!@#$&*]/g)?.join("");
  const nbSpecialCar = specialCar ? specialCar.length : 0;
  if (nbSpecialCar > 0) {
    return { score: 4 };
  }

  return { score: 3 };
};
