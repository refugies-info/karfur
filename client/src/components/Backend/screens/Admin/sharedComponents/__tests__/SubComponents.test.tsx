//@ts-nocheck
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";
import {
  TypeContenu,
  Title,
  Structure,
  StyledStatus,
  ValidateButton,
  SeeButton,
  DeleteButton,
  TabHeader,
  FilterButton,
} from "../SubComponents";
import { wrapWithProvidersAndRender } from "../../../../../../../jest/lib/wrapWithProvidersAndRender";
import { initialMockStore } from "__fixtures__/reduxStore";

let component;

test("should render TypeContenu with dispositif, detailed vue", () => {
  component = TestRenderer.create(<TypeContenu type={"dispositif"} isDetailedVue={true} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TypeContenu with demarche, detailed vue", () => {
  component = TestRenderer.create(<TypeContenu type={"demarche"} isDetailedVue={true} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TypeContenu with dispositif, not detailed vue", () => {
  component = TestRenderer.create(<TypeContenu type={"dispositif"} isDetailedVue={false} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TypeContenu with demarche, not detailed vue", () => {
  component = TestRenderer.create(<TypeContenu type={"demarche"} isDetailedVue={false} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Title with titre marque", () => {
  component = TestRenderer.create(<Title titreInformatif="titre informatif" titreMarque="titre marque" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Title without titre marque", () => {
  component = TestRenderer.create(<Title titreInformatif="titre informatif" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with null sponsor", () => {
  component = TestRenderer.create(<Structure sponsor={null} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with sponsor without nom", () => {
  const sponsor = { name: "test" };
  component = TestRenderer.create(<Structure sponsor={sponsor} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with sponsor actif", () => {
  const sponsor = { _id: "abc", nom: "test", status: "Actif" };
  component = TestRenderer.create(<Structure sponsor={sponsor} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with sponsor en attente", () => {
  const sponsor = { _id: "abc", nom: "test", status: "En attente" };
  component = TestRenderer.create(<Structure sponsor={sponsor} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with sponsor supprime", () => {
  const sponsor = { _id: "abc", nom: "test", status: "Supprimé" };
  component = TestRenderer.create(<Structure sponsor={sponsor} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus actif ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "Actif" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus en attente ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "En attente" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus Brouillon ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "Brouillon" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus Rejeté structure ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "Rejeté structure" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus En attente admin ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "En attente admin" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus Accepté structure ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "Accepté structure" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus Supprimé ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "Supprimé" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus no corresponding status ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "test" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus no corresponding with override ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "Actif", overrideColor: true, textToDisplay: "test" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus no corresponding with color ", () => {
  component = wrapWithProvidersAndRender({
    Component: StyledStatus,
    compProps: { text: "Actif", overrideColor: false, textToDisplay: "test", textColor: "textColor", color: "color" },
    reduxState: {
      ...initialMockStore,
    },
  });
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render ValidateButton disabled", () => {
  component = TestRenderer.create(<ValidateButton onClick={() => {}} disabled={true} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render ValidateButton not disabled", () => {
  component = TestRenderer.create(<ValidateButton onClick={() => {}} disabled={false} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should call onClick ValidateButton", () => {
  const onClick = jest.fn();
  component = TestRenderer.create(<ValidateButton onClick={onClick} disabled={false} />);
  component.root.findByProps({ "data-test-id": "validate-button" }).props.onClick();
  expect(onClick).toHaveBeenCalled();
});

test("should render SeeButton", () => {
  component = TestRenderer.create(<SeeButton burl="test-burl" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render DeleteButton disabled", () => {
  component = TestRenderer.create(<DeleteButton disabled={true} onClick={() => {}} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render DeleteButton not disabled", () => {
  component = TestRenderer.create(<DeleteButton disabled={false} onClick={() => {}} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should call onClick DeleteButton", () => {
  const onClick = jest.fn();
  component = TestRenderer.create(<DeleteButton onClick={onClick} disabled={false} />);
  component.root.findByProps({ "data-test-id": "delete-button" }).props.onClick();
  expect(onClick).toHaveBeenCalled();
});

test("should render FilterButton isSelected false", () => {
  component = TestRenderer.create(<FilterButton isSelected={false} onClick={() => {}} text="test" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render FilterButton isSelected true", () => {
  component = TestRenderer.create(<FilterButton isSelected={true} onClick={() => {}} text="test" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TabHeader up", () => {
  component = TestRenderer.create(<TabHeader name="test" order="true" isSortedHeader={true} sens="up" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TabHeader down", () => {
  component = TestRenderer.create(<TabHeader name="test" order="true" isSortedHeader={true} sens="down" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TabHeader no order", () => {
  component = TestRenderer.create(<TabHeader name="test" order={null} isSortedHeader={false} sens="down" />);
  expect(component.toJSON()).toMatchSnapshot();
});
