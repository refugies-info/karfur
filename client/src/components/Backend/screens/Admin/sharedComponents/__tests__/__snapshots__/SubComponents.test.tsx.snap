// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DeleteButton disabled 1`] = `
<DocumentFragment>
  .c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: not-allowed;
}

.c0:hover {
  background-color: #828282;
}

<div
    class="c0"
    data-testid="delete-button"
    disabled=""
    hovercolor="#f44336"
  >
    <div
      style="margin-bottom: 4px;"
    >
      <span
        class="icon"
      >
        <i
          style="height: 20px; line-height: 20px;"
        >
          <svg
            class="eva eva-trash"
            fill="#ffffff"
            height="20px"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <g
                data-name="trash"
              >
                <rect
                  height="24"
                  opacity="0"
                  width="24"
                />
                <path
                  d="M21 6h-5V4.33A2.42 2.42 0 0 0 13.5 2h-3A2.42 2.42 0 0 0 8 4.33V6H3a1 1 0 0 0 0 2h1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8h1a1 1 0 0 0 0-2zM10 4.33c0-.16.21-.33.5-.33h3c.29 0 .5.17.5.33V6h-4z"
                />
              </g>
            </g>
          </svg>
        </i>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render DeleteButton not disabled 1`] = `
<DocumentFragment>
  .c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: #f44336;
}

<div
    class="c0"
    data-testid="delete-button"
    hovercolor="#f44336"
  >
    <div
      style="margin-bottom: 4px;"
    >
      <span
        class="icon"
      >
        <i
          style="height: 20px; line-height: 20px;"
        >
          <svg
            class="eva eva-trash"
            fill="#ffffff"
            height="20px"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <g
                data-name="trash"
              >
                <rect
                  height="24"
                  opacity="0"
                  width="24"
                />
                <path
                  d="M21 6h-5V4.33A2.42 2.42 0 0 0 13.5 2h-3A2.42 2.42 0 0 0 8 4.33V6H3a1 1 0 0 0 0 2h1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8h1a1 1 0 0 0 0-2zM10 4.33c0-.16.21-.33.5-.33h3c.29 0 .5.17.5.33V6h-4z"
                />
              </g>
            </g>
          </svg>
        </i>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render FilterButton isSelected false 1`] = `
<DocumentFragment>
  .c0 {
  background: #ffffff;
  color: #212121;
  border-radius: 12px;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  padding: 15px;
  width: fit-content;
  margin-right: 8px;
  cursor: pointer;
  height: fit-content;
}

<div
    class="c0"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`should render FilterButton isSelected true 1`] = `
<DocumentFragment>
  .c0 {
  background: #212121;
  color: #ffffff;
  border-radius: 12px;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  padding: 15px;
  width: fit-content;
  margin-right: 8px;
  cursor: pointer;
  height: fit-content;
}

<div
    class="c0"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`should render SeeButton 1`] = `
<DocumentFragment>
  .c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: #212121;
}

<div
    class="c0"
    hovercolor="#212121"
  >
    <a
      href="test-burl"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        style="margin-bottom: 4px;"
      >
        <span
          class="icon"
        >
          <i
            style="height: 20px; line-height: 20px;"
          >
            <svg
              class="eva eva-eye"
              fill="#ffffff"
              height="20px"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 2"
              >
                <g
                  data-name="eye"
                >
                  <rect
                    height="24"
                    opacity="0"
                    width="24"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="1.5"
                  />
                  <path
                    d="M21.87 11.5c-.64-1.11-4.16-6.68-10.14-6.5-5.53.14-8.73 5-9.6 6.5a1 1 0 0 0 0 1c.63 1.09 4 6.5 9.89 6.5h.25c5.53-.14 8.74-5 9.6-6.5a1 1 0 0 0 0-1zm-9.87 4a3.5 3.5 0 1 1 3.5-3.5 3.5 3.5 0 0 1-3.5 3.5z"
                  />
                </g>
              </g>
            </svg>
          </i>
        </span>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`should render Structure with null sponsor 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #f44336;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      color="#f44336"
    />
    <div
      class="c2"
    >
      Sans structure
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render Structure with sponsor actif 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #4caf50;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      color="#4caf50"
    />
    <div
      class="c2"
    >
      test
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render Structure with sponsor en attente 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #FF9800;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      color="#FF9800"
    />
    <div
      class="c2"
    >
      test
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render Structure with sponsor supprime 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #f44336;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      color="#f44336"
    />
    <div
      class="c2"
    >
      test
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render Structure with sponsor without nom 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #f44336;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      color="#f44336"
    />
    <div
      class="c2"
    >
      Sans structure
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus Accepté structure  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #FF9800;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
    class="c0"
    color="#FF9800"
    textcolor="#ffffff"
  >
    Accepté
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus Brouillon  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #FFEB3B;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #212121;
}

<div
    class="c0"
    color="#FFEB3B"
    textcolor="#212121"
  >
    Brouillon
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus En attente admin  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #8BC34A;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
    class="c0"
    color="#8BC34A"
    textcolor="#ffffff"
  >
    Relecture en cours
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus Rejeté structure  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #F44336;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
    class="c0"
    color="#F44336"
    textcolor="#ffffff"
  >
    Rejeté
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus Supprimé  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #F44336;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
    class="c0"
    color="#F44336"
    textcolor="#ffffff"
  >
    Supprimé
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus actif  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #4CAF50;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
    class="c0"
    color="#4CAF50"
  >
    Publié
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus en attente  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #FF9800;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
    class="c0"
    color="#FF9800"
    textcolor="#ffffff"
  >
    En attente
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus no corresponding status  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #0421b1;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
    class="c0"
    color="#0421b1"
    textcolor="#ffffff"
  >
    Nouveau !
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus no corresponding with color  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: color;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: textColor;
}

<div
    class="c0"
    color="color"
    textcolor="textColor"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`should render StyledStatus no corresponding with override  1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #828282;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
    class="c0"
    color="#828282"
    textcolor="#ffffff"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`should render TabHeader down 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  font-weight: bold;
  cursor: pointer;
}

<div
    class="c0"
    order="1"
  >
    test
    <span
      class="icon"
    >
      <i
        style="height: 20px; line-height: 20px;"
      >
        <svg
          class="eva eva-chevron-down"
          fill="#212121"
          height="20px"
          viewBox="0 0 24 24"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <g
              data-name="chevron-down"
            >
              <rect
                height="24"
                opacity="0"
                width="24"
              />
              <path
                d="M12 15.5a1 1 0 0 1-.71-.29l-4-4a1 1 0 1 1 1.42-1.42L12 13.1l3.3-3.18a1 1 0 1 1 1.38 1.44l-4 3.86a1 1 0 0 1-.68.28z"
              />
            </g>
          </g>
        </svg>
      </i>
    </span>
  </div>
</DocumentFragment>
`;

exports[`should render TabHeader no order 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  font-weight: bold;
  cursor: inherit;
}

<div
    class="c0"
    order="0"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`should render TabHeader up 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  font-weight: bold;
  cursor: pointer;
}

<div
    class="c0"
    order="1"
  >
    test
    <span
      class="icon"
    >
      <i
        style="height: 20px; line-height: 20px;"
      >
        <svg
          class="eva eva-chevron-up"
          fill="#212121"
          height="20px"
          viewBox="0 0 24 24"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <g
              data-name="chevron-up"
            >
              <rect
                height="24"
                opacity="0"
                transform="rotate(180 12 12)"
                width="24"
              />
              <path
                d="M16 14.5a1 1 0 0 1-.71-.29L12 10.9l-3.3 3.18a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.42l4-3.86a1 1 0 0 1 1.4 0l4 4a1 1 0 0 1 0 1.42 1 1 0 0 1-.69.28z"
              />
            </g>
          </g>
        </svg>
      </i>
    </span>
  </div>
</DocumentFragment>
`;

exports[`should render Title with titre marque 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  text-align: left;
  font-style: normal;
  font-size: 16px;
  line-height: 20px;
  max-width: 370px;
  flex: 1;
  cursor: pointer;
}

<div
    class="c0"
  >
    <b>
      titre informatif
    </b>
    <span>
      avec titre marque
    </span>
  </div>
</DocumentFragment>
`;

exports[`should render Title without titre marque 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  text-align: left;
  font-style: normal;
  font-size: 16px;
  line-height: 20px;
  max-width: 370px;
  flex: 1;
  cursor: pointer;
}

<div
    class="c0"
  >
    <b>
      titre informatif
    </b>
  </div>
</DocumentFragment>
`;

exports[`should render TypeContenu with demarche, detailed vue 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #ffffff;
  background-color: #212121;
  padding: 8px;
  border-radius: 6px;
  width: fit-content;
  cursor: pointer;
}

<div
    class="c0"
  >
    Démarche
  </div>
</DocumentFragment>
`;

exports[`should render TypeContenu with demarche, not detailed vue 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #212121;
  background-color: #ffffff;
  padding: 8px;
  border-radius: 6px;
  width: fit-content;
  cursor: pointer;
}

<div
    class="c0"
  >
    Démarche
  </div>
</DocumentFragment>
`;

exports[`should render TypeContenu with dispositif, detailed vue 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #ffffff;
  background-color: #212121;
  padding: 8px;
  border-radius: 6px;
  width: fit-content;
  cursor: pointer;
}

<div
    class="c0"
  >
    Dispositif
  </div>
</DocumentFragment>
`;

exports[`should render TypeContenu with dispositif, not detailed vue 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #ffffff;
  background-color: #212121;
  padding: 8px;
  border-radius: 6px;
  width: fit-content;
  cursor: pointer;
}

<div
    class="c0"
  >
    Dispositif
  </div>
</DocumentFragment>
`;

exports[`should render ValidateButton disabled 1`] = `
<DocumentFragment>
  .c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: not-allowed;
}

.c0:hover {
  background-color: #828282;
}

<div
    class="c0"
    data-testid="validate-button"
    disabled=""
    hovercolor="#4caf50"
  >
    <div
      style="margin-bottom: 4px;"
    >
      <span
        class="icon"
      >
        <i
          style="height: 20px; line-height: 20px;"
        >
          <svg
            class="eva eva-checkmark-outline"
            fill="#ffffff"
            height="20px"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <g
                data-name="checkmark"
              >
                <rect
                  height="24"
                  opacity="0"
                  width="24"
                />
                <path
                  d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                />
              </g>
            </g>
          </svg>
        </i>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render ValidateButton not disabled 1`] = `
<DocumentFragment>
  .c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: #4caf50;
}

<div
    class="c0"
    data-testid="validate-button"
    hovercolor="#4caf50"
  >
    <div
      style="margin-bottom: 4px;"
    >
      <span
        class="icon"
      >
        <i
          style="height: 20px; line-height: 20px;"
        >
          <svg
            class="eva eva-checkmark-outline"
            fill="#ffffff"
            height="20px"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <g
                data-name="checkmark"
              >
                <rect
                  height="24"
                  opacity="0"
                  width="24"
                />
                <path
                  d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                />
              </g>
            </g>
          </svg>
        </i>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
